torchgeo/datasets/seasonet.py

```diff
@@ -0,0 +1,456 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+
+"""SeasoNet dataset."""
+
+import os
+import random
+from typing import Callable, Collection, Iterable, Optional
```

Comment:
Most of these can be imported from `collections.abc`, which is why pyupgrade tests are failing